### PR TITLE
Update Contribution Guide - Bicep

### PR DESCRIPTION
**Pull request by eriqua**
# Overview/Summary

Update Bicep Contribution Guide

## This PR fixes/adds/changes/removes

1. Update tooling
2. Add contribution composition guidelines
3. Add versioning

### Breaking Changes

N/A

## As part of this Pull Request I have

- [x] Read the Contribution Guide and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Verified-Modules/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Verified-Modules/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Verified-Modules/)
- [x] Ensured PR tests are passing
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: docs/content/contributing/bicep/_index.md [hunk 1]
```diff
@@ -34,10 +34,24 @@ Before you start contributing to the AVM, it is **highly recommended** that you
...
+
+  {{< hint type=note >}}
+
+  Must be manually kept up-to-date.
+
+  {{< /hint >}}
+
 - [Visual Studio Code](https://code.visualstudio.com/download)
   - [Bicep extension for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-bicep)
-  - [EditorConfig for VS Code](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig)
```
Comment (jtracey93) [inline docs/content/contributing/bicep/_index.md:40]: @eriqua We need this one. As it's used for linting and enforcing spaces, tabs and white space etc in the repo

**Review by jtracey93 — CHANGES_REQUESTED** [review]

File: docs/content/contributing/bicep/_index.md (hunk 1 above)
Comment (eriqua) [reply docs/content/contributing/bicep/_index.md:40]: Removed as duplicated below in the _recommended_ tools. If you vouch for that to be required I'll remove it from the next list instead

Comment (jtracey93) [reply docs/content/contributing/bicep/_index.md:40]: I would like it recommended as folks Will get linting errors otherwise more commonly.

Comment (eriqua) [reply docs/content/contributing/bicep/_index.md:40]: I agree the extension is useful. We can either
1) List that in the recommended tools and remove it from required tools (proposed change) -OR-
2) List that in the required tools and remove it from recommended tools 

I don't have a strong preference, but I'd avoid having the same extension listed twice as it is currently.

Comment (jtracey93) [reply docs/content/contributing/bicep/_index.md:40]: Okay lets leave as recommended 👍

**Review by jtracey93 — APPROVED** [review]
LGTM

**Pull request merged**